docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+root-overlay              OK        Filesystem
+var-log                   OK        Filesystem
+routeCheck                OK        Program
+dualtorNeighborCheck      OK        Program
+container_memory_gnmi     OK        Program
+container_eventd          OK        Program
+database:redis            OK        Process
+bgp:zebra                 OK        Process
+bgp:staticd               OK        Process
+teamd:tlm_teamd           OK        Process
+eventd:eventd             OK        Process
+dpu-pdsagent              OK        UserDefine
+dpu-pciemgrd              OK        UserDefine
+dpu-eth_Uplink1/1_status  OK        UserDefine
+dpu-pcie_link             OK        UserDefine
+
+On Switch: “show system-health detail SWITCH"
+Output: Same as “show system-health detail"
+
+On Switch: “show system-health detail all"
+Output: Same as “show system-health detail SWITCH" plus the DPUs output
+
+```
+#### Pass/Fail Criteria
+ *  A proper output should be delayed without any error
+
+### 1.13 Check the show system-health dpu DPUx CLI on the SWITCH
+
+#### Steps
+ * Issue this CLI “show system-health dpu DPU0" on the switch
+ * Verify the DPU0 state information
+ * Check the midplane state, control-plane state, data-plane state
+ * Alter the control-plane state, data-plane of the DPU by causing a container failure and pipeline failure respectively
+ * Verify the state changes get reflected
+ * TBD: midplane state
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch: “show system-health dpu DPU0"
+
+Name    ID    Oper-Status    State-Detail             State-Value    Time               Reason
+------  ----  -------------  -----------------------  -------------  -----------------  --------------------------------------------------------------------------------------------------------------------------
+DPU0    0     Online         dpu_midplane_link_state  UP             20240605 02:01:13  INTERNAL-MGMT : admin state - UP, oper_state - UP, status - OK, HOST-MGMT : admin state - UP, oper_state - UP, status - OK
+                             dpu_control_plane_state  UP             20240605 02:01:13  All containers are up and running, host-ethlink-status: Uplink1/1 is UP
+                             dpu_data_plane_state     UP             20240605 00:03:55  DPU container named polaris is running, pdsagent running : OK, pciemgrd running : OK
+
+```
+#### Pass/Fail Criteria
+ *  A proper output should be delayed without any error
+
+### 1.14 Check the cold startup of the DPUs and SWITCH
+
+#### Steps
+ * Set the config_db.json such that some DPUs are powered down
+ * Cold start the system with the config
+ * Verify if the expected DPUs  are ON and the remaining are OFF and the  admin_status is set properly
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+Once the all the containers are up, (give enough time for the DPUs to boot up as well) and the system is stable
+On Switch: “show platform inventory"
```

Comment:
The inventory command output shown here is vendor-specific and should be generalized to meet the requirements and capabilities of all vendors. The 'Route Processors' seems to be the same as 'CHASSIS,' so it can be removed. The 'Sled Cards' is Cisco-specific, so it should be optional or removed from the command output.

Additionally, where should the 'Version' information be taken from? It is not part of the vendor API. In the vendor API, we can only query the serial number of the module.

Reply:
The version information is vendor specific and this output doesn't mean that all the columns are must. As you mentioned we can only query the serial number of the module using the vendor API.

Reply:
cleaned up some, but left the version column and once again it is not a requirement.